src/main/java/org/reaktivity/nukleus/http2/internal/routable/stream/Http2Stream.java

```diff
@@ -0,0 +1,383 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.reaktivity.nukleus.http2.internal.routable.stream;
+
+import org.agrona.DirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.reaktivity.nukleus.http2.internal.routable.Route;
+import org.reaktivity.nukleus.http2.internal.routable.Target;
+import org.reaktivity.nukleus.http2.internal.types.HttpHeaderFW;
+import org.reaktivity.nukleus.http2.internal.types.ListFW;
+import org.reaktivity.nukleus.http2.internal.types.stream.HpackContext;
+import org.reaktivity.nukleus.http2.internal.types.stream.HpackHeaderFieldFW;
+import org.reaktivity.nukleus.http2.internal.types.stream.HpackHuffman;
+import org.reaktivity.nukleus.http2.internal.types.stream.HpackLiteralHeaderFieldFW;
+import org.reaktivity.nukleus.http2.internal.types.stream.HpackStringFW;
+import org.reaktivity.nukleus.http2.internal.types.stream.Http2DataFW;
+import org.reaktivity.nukleus.http2.internal.types.stream.Http2ErrorCode;
+import org.reaktivity.nukleus.http2.internal.types.stream.Http2FrameFW;
+import org.reaktivity.nukleus.http2.internal.types.stream.Http2FrameType;
+import org.reaktivity.nukleus.http2.internal.types.stream.Http2HeadersFW;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.reaktivity.nukleus.http2.internal.types.stream.HpackLiteralHeaderFieldFW.LiteralType.INCREMENTAL_INDEXING;
+
+class Http2Stream
+{
+
+    enum State
+    {
+        IDLE,
+        RESERVED_LOCAL,
+        RESERVED_REMOTE,
+        OPEN,
+        HALF_CLOSED_LOCAL,
+        HALF_CLOSED_REMOTE,
+        CLOSED
+    }
+
+    private final SourceInputStreamFactory.SourceInputStream connection;
+    private final int http2StreamId;
+    private final long targetId;
+    private Route route;
+    private State state;
+
+    Http2Stream(SourceInputStreamFactory.SourceInputStream connection, int http2StreamId, long targetId)
+    {
+        this.connection = connection;
+        this.http2StreamId = http2StreamId;
+        this.targetId = targetId;
+        this.state = State.IDLE;
+    }
+
+    void decode(Http2FrameFW http2RO)
+    {
+        switch (state)
+        {
+            case IDLE:
+                inIdle(http2RO);
+                break;
+            case RESERVED_LOCAL:
+                inReservedLocal(http2RO);
+                break;
+            case RESERVED_REMOTE:
+                inReservedRemote(http2RO);
+                break;
+            case OPEN:
+                inOpen(http2RO);
+                break;
+            case HALF_CLOSED_LOCAL:
+                inHalfClosedLocal(http2RO);
+                break;
+            case HALF_CLOSED_REMOTE:
+                inHalfClosedRemote(http2RO);
+                break;
+            case CLOSED:
+                inClosed(http2RO);
+                break;
+        }
```

Comment:
The state machine is not implemented completely. I will address this in future PR.